src/Translator/doc/index.rst

```diff
@@ -0,0 +1,101 @@
+Symfony UX Translator
+=====================
+
+Symfony UX Translator is a Symfony bundle providing the same mechanism as `Symfony Translator`_
+in JavaScript, in Symfony applications. It is part of `the Symfony UX initiative`_.
+
+The `ICU Message Format`_ is also supported.
+
+Installation
+------------
+
+Before you start, make sure you have `Symfony UX configured in your app`_.
+
+Install this bundle using Composer and Symfony Flex:
+
+.. code-block:: terminal
+
+    $ composer require symfony/ux-translator
+
+    # Don't forget to install the JavaScript dependencies as well and compile
+    $ npm install --force
+    $ npm run watch
+
+    # or use yarn
+    $ yarn install --force
+    $ yarn watch
+
+Usage
+-----
+
+Initializing the JavaScript translator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The JavaScript translator must be initialized in your application.
+This can be done by calling the Twig function ``initialize_js_translator()``:
+
+.. code-block:: twig
+
+    {# templates/base.html.twig #}
+
+    {% block javascripts %}
+        {{ initialize_js_translator() }}
+    {% endblock %}
+
+
+Loading translations for JavaScript
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To load translations for JavaScript, you need to to call the Twig function ``load_translations_for_js()`` with the
+explicit list of translations keys you want to load:
+
+.. code-block:: twig
+
+    {# templates/another/template.html.twig #}
+
+    {% extends 'base.html.twig' %}
+
+    {% block javascripts %}
+        {{ parent() }}
+        {{ load_translations_for_js([
+            'app.translation_1',
+            'app.translation_2',
+            'app.translation_3',
+        ]) }}
+
+        {% if ... %}
+            {{ load_translations_for_js([
+                'app.translation_4',
+                'app.translation_5',
+            ]) }}
+        {% endif %}
+    {% endblock %}
+
+
+.. note::
+    Specifying each translations keys one by one can be annoying, but it ensures that not every translations are loaded,
+    which could make your HTML page or your JavaScript files heavier.
```

Comment:
Could/should there also be a way to dump an entire "domain" - e.g. maybe the user organizes things into smaller domains by feature, and they want to dump everything?

Reply:
The documentation is outdated, I'm rewriting the whole package (in order to have a better DX) and this won't apply anymore, since every translations from all possible domains will be dump to the filesystem.

And we won't have heavy JS bundles after compiling assets, thanks to the three-shaking.